test-interface/src/main/scala/scala/scalanative/testinterface/NativeRPC.scala

```diff
@@ -0,0 +1,80 @@
+package scala.scalanative.testinterface
+
+import java.io.{DataInputStream, DataOutputStream}
+import java.net.Socket
+import scala.annotation.tailrec
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.scalanative.testinterface.common.RPCCore
+import scala.util.{Failure, Success, Try}
+
+/** Native RPC Core. */
+private[testinterface] object NativeRPC extends RPCCore {
+  Com.init(handleMessage)
+
+  def loop(clientSocket: Socket): Int = {
+    Com.socket = clientSocket
+    Com.loop()
+  }
+
+  override protected def send(msg: String): Unit = Com.send(msg)
+
+  private object Com {
+    private[this] val messageQueue =
+      scala.collection.mutable.Queue.empty[String]
+    private var messageHandler: String => Unit = _
+    private[NativeRPC] var socket: Socket      = _
+
+    private lazy val inStream  = new DataInputStream(socket.getInputStream)
+    private lazy val outStream = new DataOutputStream(socket.getOutputStream)
+
+    def init(onReceive: String => Unit): Unit = {
+      if (messageHandler != null) sys.error("Com already initialized")
+      messageHandler = onReceive
+      messageQueue.foreach(onReceive)
```

Comment:
If fact since `NativeRPC` is now class taking `Socket` as constructor argument whole `init` method is redundant. MessageQueue would never take any item, also there's no reason to pass `onReceive` param since `NativeRPC` extends `RPCCore` containing message handler that would be used anyway